### ISSUE TITLE
[aon_timer] Consider requirements for clk_aon_i vs clk_i

### ISSUE THREAD

**Issue by GregAC** [lowRISC/opentitan#19059]
The current aon_timer testbench has a limitation so we must start clk_aon_i before clk_i can start (https://github.com/lowRISC/opentitan/issues/6821). This is not an issue for earlgrey as the aon clock is always active before other clocks (the FSM in `pwrmgr` that enables the fast clocks is clocked from the aon clock).

We should consider either making this an explicit design constraint or fixing the limitation to allow the fast clock to start with the aon clock in the aon_timer testbench.

I'd favour the former as it seems like a reasonable design constraint.

**Comment by matutem** [comment]
I agree with making this an explicit design constraint for the aon_timer testbench.

**Comment by andreaskurth** [comment]
This constraint sounds sensible to me as well. So we need to update the aon_timer documentation/specification and constrain the DV environment appropriately? Or does the DV environment already do this correctly?

**Comment by andreaskurth** [comment]
Assigning @GregAC to please take a look and propose a resolution -- feel free to delegate the implementation ofc

**Comment by andreaskurth** [comment]
By @GregAC: This seems to be a doc-only issue. We don't care about starting the other clocks before the AON clock for Earlgrey. But we should confirm this with the PD team.

By @rswarbrick: We should add an assertion / constraint in the chip-level testbench to ensure this always holds.

**Comment by andreaskurth** [comment]
Confirmed by PD team that clk_aon_i is always active before clk_i
